### PR TITLE
docs(workflow): add rebase use case

### PR DESCRIPTION
**Pull request by jd**
Fixes MRGFY-2719

Change-Id: I780941751c6af4f5000b9c81938e841963d1c180

### CODE REVIEW

File: src/content/workflow/rebase.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+is helpful when you want to ensure your changes are built upon the most recent
+commits of the branch you're targeting.
+
+Rebasing a pull request in GitHub allows you to update the entire commit
+history of a branch onto its base branch, while the base branch received new
+commits.
+
+Mergify allows to rebase pull requests in two different ways:
+
+- Either by using a the [`rebase` command](/commands/rebase) which rebases the
```
Comment (Syffe) [inline src/content/workflow/rebase.mdx:21]:
```suggestion
- Either by using the [`rebase` command](/commands/rebase) which rebases the
```
or
```suggestion
- Either by using a [`rebase` command](/commands/rebase) which rebases the
```

File: src/content/workflow/rebase.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  pull request immediately;
+
+- Either by using an automation rule that automatically triggers a rebase
+  when a pull request meets conditions.
+
+## Rebasing a PR from GitHub
+
+To instantly rebase a pull request without leaving GitHub, uses the Mergify
```
Comment (Syffe) [inline src/content/workflow/rebase.mdx:29]:
```suggestion
To instantly rebase a pull request without leaving GitHub, use the Mergify
```

File: src/content/workflow/rebase.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+With this rule, as soon as the pull request is more than 10 commits behind the
+`main` branch, Mergify automatically rebases the pull request which therefore
+retriggers the CI. This is useful to make sure that you don't merge pull
+request that are too much out-of-date.
+
+<Warning>
+  If your goal is to only merge pull request that are rebased and retested
+  before being merged, you are actually looking for a [merge
```
Comment (Syffe) [inline src/content/workflow/rebase.mdx:103]:
```suggestion
  If your goal is to only merge pull request that are rebased and retested, you are actually looking for a [merge
```
WDYT ?

File: src/content/index.mdx [hunk 4]
```diff
@@ -83,14 +83,12 @@ and adapt it to your own needs.
     path="/workflow/dismiss-reviews"
     icon={<IoIosRemoveCircleOutline/>}
   />
-  {/*
   <Docset
-    title="Posting Comments"
+    title="Rebasing Pull Requests"
     description="Post comments to help your teammates."
```
Comment (DouglasBlackwood) [inline src/content/index.mdx:88]: Wrong description

File: src/content/workflow/rebase.mdx [hunk 5]
```diff
@@ -0,0 +1,126 @@
+---
+title: Rebasing Pull Requests
+description: How to automatically delete head branches.
```
Comment (DouglasBlackwood) [inline src/content/workflow/rebase.mdx:3]: Wrong description?

File: src/content/workflow/rebase.mdx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+This command triggers a `git rebase` operation and force-push the rebased
+branch.
+
+import RebaseAdvantage from '../commands/_rebase_advantages.mdx';
+
+<RebaseAdvantage />
+
+## Automatically Rebasing PRs
+
+You might be tempted to rebase pull request automatically when they are behind
```
Comment (DouglasBlackwood) [inline src/content/workflow/rebase.mdx:47]:
```suggestion
You might be tempted to rebase pull requests automatically when they are behind
```

File: src/content/workflow/rebase.mdx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+their base branch, such as `main` or `master`. The [`rebase`
+action](/workflow/actions/rebase) is a great candidate to help in such a case.
+
+You can write an automation rule that does exactly this for you:
+
+```yaml
+pull_request_rules:
+  - name: continuously rebase pull request when it's labeled with `rebase`
```
Comment (DouglasBlackwood) [inline src/content/workflow/rebase.mdx:55]:
```suggestion
  - name: continuously rebase pull requests when it's labeled with `rebase`
```

File: src/content/workflow/rebase.mdx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+Note that this rebases the pull request as long as it has the label.
+
+If you wanted to rebase only once, you would also need to remove the label once
+the rebase is done:
+
+```yaml
+pull_request_rules:
+  - name: rebase pull request once when it's labeled with `rebase`
```
Comment (DouglasBlackwood) [inline src/content/workflow/rebase.mdx:69]:
```suggestion
  - name: rebase pull requests once when it's labeled with `rebase`
```

File: src/content/workflow/rebase.mdx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+      rebase:
+```
+
+With this rule, as soon as the pull request is more than 10 commits behind the
+`main` branch, Mergify automatically rebases the pull request which therefore
+retriggers the CI. This is useful to make sure that you don't merge pull
+request that are too much out-of-date.
+
+<Warning>
+  If your goal is to only merge pull request that are rebased and retested
```
Comment (DouglasBlackwood) [inline src/content/workflow/rebase.mdx:102]:
```suggestion
  If your goal is to only merge pull requests that are rebased and retested
```

File: src/content/workflow/rebase.mdx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+  before being merged, you are actually looking for a [merge
+  queue](/merge-queue/intro). A merge queue makes sure each pull request is
+  up to date and retested before being merged, while avoiding to update all
+  pull request at once.
```
Comment (DouglasBlackwood) [inline src/content/workflow/rebase.mdx:106]:
```suggestion
  pull requests at once.
```

File: src/content/workflow/rebase.mdx (hunk 3 above)
Comment (jd) [reply src/content/workflow/rebase.mdx:103]: I think I prefer the explicit version :)

**Review by DouglasBlackwood — APPROVED** [review]

**Review by Syffe — APPROVED** [review]

**Pull request merged**